stdlib/public/core/StringStorageBridge.swift

```diff
@@ -84,12 +87,52 @@ extension _AbstractStringStorage {
   internal func _cString(encoding: UInt) -> UnsafePointer<UInt8>? {
     switch (encoding, isASCII) {
     case (_cocoaASCIIEncoding, true),
+         (_cocoaMacRomanEncoding, true),
          (_cocoaUTF8Encoding, _):
       return unsafe start
     default:
       return _cocoaCStringUsingEncodingTrampoline(self, encoding)
     }
   }
+  
+  @_effects(readonly)
+  internal func _maximumLengthOfBytes(using encoding: UInt) -> Int {
+    switch encoding {
+    case _cocoaASCIIEncoding, _cocoaMacRomanEncoding:
+      if isASCII) {
+        return count
+      }
+      return UTF16Count // This is weird but it's what NSString does
+    case _cocoaUTF8Encoding:
+      return count
+    case _cocoaUTF16Encoding:
+      return UTF16Count
```

Comment:
Hmm. We should be able to do better than this here actually. Let me try to figure out what the worst case scenario is for utf8 -> utf16 length expansion